pkg/util/tracing/tracer.go

```diff
@@ -401,6 +386,27 @@ func (t *Tracer) startSpanGeneric(
 		}
 	}
 
+	// Set initial tags.
```

Comment:
This movement was necessary to fix a bug: we were linking the shadow span after adding baggage to the parent, so the shadow span never received them. Erroneously passed before my other fixes here.